src/graph.rs

```diff
@@ -354,6 +302,78 @@ impl From<String> for Value {
     }
 }
 
+impl fmt::Display for Value {
+    fn fmt(&self, f: &mut fmt::Formatter) -> fmt::Result {
+        match self {
+            Value::Null => write!(f, "#null"),
+            Value::Boolean(value) => write!(f, "{}", value),
+            Value::Integer(value) => write!(f, "{}", value),
+            Value::String(value) => write!(f, "\"{}\"", value),
+            Value::List(values) => {
+                write!(f, "[")?;
+                for elem in values {
+                    write!(f, "{},", elem)?;
+                }
+                write!(f, "]")
+            }
+            Value::Set(values) => {
+                write!(f, "{{")?;
+                for elem in values {
+                    write!(f, "{},", elem)?;
+                }
+                write!(f, "}}")
+            }
+            Value::SyntaxNode(value) => <dyn fmt::Display>::fmt(value, f),
+            Value::GraphNode(value) => <dyn fmt::Display>::fmt(value, f),
+        }
+    }
+}
+
+impl DisplayWithGraph for Value {
```

Comment:
This trait implementation replaces the method that was implemented directly on `Value`.